plutus-core/plc/Main.hs

```diff
@@ -759,44 +772,64 @@ printBudgetStateTally (Cek.CekExTally costs) = do
   putStrLn $ "Error      " ++ pbudget Cek.BError
   putStrLn $ "Builtin    " ++ pbudget Cek.BBuiltin
   putStrLn ""
-  putStrLn $ "AST        " ++ pbudget Cek.BAST
-  putStrLn $ "compute    " ++ printf "%-20s" (budgetToString totalComputeSteps)
-  putStrLn $ "BuiltinApp " ++ budgetToString (mconcat (map snd builtinsAndCosts))
+  putStrLn $ "startup    " ++ pbudget Cek.BStartup
+  putStrLn $ "compute    " ++ printf "%-20s" (budgetToString totalComputeCost)
+  putStrLn $ "AST nodes  " ++ printf "%15d" (UPLC.termSize term)
   putStrLn ""
-  traverse_ (\(b,cost) -> putStrLn $ printf "%-20s %s" (show b) (budgetToString cost :: String)) builtinsAndCosts
-      where
-        get k =
+  putStrLn $ "BuiltinApp " ++ budgetToString builtinCosts
+  case model of
+    Default ->
+        do
+  -- 1e9*(0.200  + 0.0000725 * totalComputeSteps + builtinExeTimes/1000)  putStrLn ""
+          putStrLn ""
+          traverse_ (\(b,cost) -> putStrLn $ printf "%-20s %s" (show b) (budgetToString cost :: String)) builtinsAndCosts
+          putStrLn ""
+          putStrLn $ "Total budget spent: " ++ printf (budgetToString totalCost)
+          putStrLn $ "Predicted execution time: " ++ (formatTime_picoseconds totalTime)
+    Unit -> pure ()
+  where
+        getSpent k =
             case H.lookup k costs of
               Just v  -> v
               Nothing -> ExBudget 0 0
         allNodeTags = [Cek.BConst, Cek.BVar, Cek.BLamAbs, Cek.BApply, Cek.BDelay, Cek.BForce, Cek.BError, Cek.BBuiltin]
-        totalComputeSteps = mconcat $ map get allNodeTags  -- Depends on the fact that we have a unit cost for each AST node type
-        budgetToString (ExBudget (ExCPU cpu) (ExMemory mem)) = printf "%10d  %10d" (toInteger cpu) (toInteger mem) :: String
-        pbudget k = budgetToString $ get k
+        totalComputeCost = mconcat $ map getSpent allNodeTags  -- For unitCekCosts this will be the total number of compute steps
```

Comment:
Nit: `mconcat . map f` can be spelled as `foldMap f` (although for certains monoids `mconcat` can be faster than `fold`).